src/controls/control.class.ts

```diff
@@ -1,372 +1,392 @@
-     * @param {fabric.Object} fabricObject on which the control is displayed
-     * @param {fabric.Control} control control for which the action handler is being asked
-     * @return {Function} the action handler
-     */
-    getMouseUpHandler: function (/* eventData, fabricObject, control */) {
-      return this.mouseUpHandler;
-    },
+  /**
+   * Returns control cursorStyle for css using cursorStyle. If you need a more elaborate
+   * function you can pass one in the constructor
+   * the cursorStyle property
+   * @param {Event} eventData the native mouse event
+   * @param {Control} control the current control ( likely this)
+   * @param {FabricObject} object on which the control is displayed
+   * @return {String}
+   */
+  cursorStyleHandler(
+    eventData: TPointerEvent,
+    control: Control,
+    fabricObject: FabricObject
+  ) {
+    return control.cursorStyle;
+  }
 
-    /**
-     * Returns control cursorStyle for css using cursorStyle. If you need a more elaborate
-     * function you can pass one in the constructor
-     * the cursorStyle property
-     * @param {Event} eventData the native mouse event
-     * @param {fabric.Control} control the current control ( likely this)
-     * @param {fabric.Object} object on which the control is displayed
-     * @return {String}
-     */
-    cursorStyleHandler: function (eventData, control /* fabricObject */) {
-      return control.cursorStyle;
-    },
+  /**
+   * Returns the action name. The basic implementation just return the actionName property.
+   * @param {Event} eventData the native mouse event
+   * @param {Control} control the current control ( likely this)
+   * @param {FabricObject} object on which the control is displayed
+   * @return {String}
+   */
+  getActionName(
+    eventData: TPointerEvent,
+    control: Control,
+    fabricObject: FabricObject
+  ) {
+    return control.actionName;
+  }
 
-    /**
-     * Returns the action name. The basic implementation just return the actionName property.
-     * @param {Event} eventData the native mouse event
-     * @param {fabric.Control} control the current control ( likely this)
-     * @param {fabric.Object} object on which the control is displayed
-     * @return {String}
-     */
-    getActionName: function (eventData, control /* fabricObject */) {
-      return control.actionName;
-    },
+  /**
+   * Returns controls visibility
+   * @param {FabricObject} object on which the control is displayed
+   * @param {String} controlKey key where the control is memorized on the
+   * @return {Boolean}
+   */
+  getVisibility(fabricObject: FabricObject, controlKey: string) {
+    // @ts-expect-error TODO remove directive once fixed
+    const objectVisibility = fabricObject._controlsVisibility;
+    if (
+      objectVisibility &&
+      typeof objectVisibility[controlKey] !== 'undefined'
+    ) {
+      return objectVisibility[controlKey];
+    }
+    return this.visible;
```

Comment:
i think this is one of the few cases that is really equivalent to
```js
return objectVisibility?.[controlKey] ?? this.visible
```

Reply:
`objectVisibility?.[controlKey] ` 
this is new syntax for me